cmd/metrics-api/metrics-api.go

```diff
@@ -20,48 +20,19 @@ package main
 import (
 	"net/http"
 
+	"github.com/aerogear/aerogear-app-metrics/internal/setup"
 	"github.com/aerogear/aerogear-app-metrics/pkg/config"
-	"github.com/aerogear/aerogear-app-metrics/pkg/dao"
-	"github.com/aerogear/aerogear-app-metrics/pkg/mobile"
-	"github.com/aerogear/aerogear-app-metrics/pkg/web"
 	"github.com/rs/cors"
 	log "github.com/sirupsen/logrus"
 )
 
 func main() {
-
 	config := config.GetConfig()
+	metricsDao := setup.InitDao(config)
```

Comment:
Some minor points:

1. This has created an unnecessary dependency between the setup package and the config package.
2. Non db related options are being passed into a function called `InitDao`

I think the `config` struct should stay internal and the config package should only be depended upon at a top level. I would change this to `setup.InitDao(config.DBConnectionString, config.DBMaxConnections)` to avoid such a dependency. Would you agree?

Reply:
Done!